### PR TITLE
ZURI CHAT - Fix  the Session signout link on the Account Profile settings and route properly

### PR DESCRIPTION
**Pull request by Chisquare7**
Linear Ticket link: https://linear.app/team-brainbox/issue/FRO-196/zuri-chat

# Fixes Issue/Linear Ticket
**My PR Fixes ID-linear_ticket_number_FRO-196**

# Changes proposed
Worked on fixing the Session signout link on the Account Profile settings and route properly

## What were you told to do?
ZURI CHAT - Fix  the Session signout link on the Account Profile settings and route properly

## What did you do?
- I moved Session signout files from the old src folder to new src folder in the protected folder
- I fixed all the import errors and paths of other files associated with Session signout files
- I fixed the routing links for Session signout file in the App.jsx file in the new src folder
- I moved necessary and equivalent assets from old src folder to new src folder

# Check List (Check all the applicable boxes)

- [X] My code follows the code style of this project.
- [X] This PR does not contain plagiarized content.
- [X] The title and description of the PR is clear and explains the approach.
- [X] I am making a pull request against the **dev branch** (left side).
- [X] My commit messages styles matches our requested structure.
- [X] My code additions will fail neither code linting checks nor unit test.
- [X] I am only making changes to files I was requested to.

# Screenshots/ Videos

Before changes:

After changes:
image

### CODE REVIEW

File: packages/main/src/components/sessions-signout/AllSessionSignOut.jsx [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+import SignOutActionList from "./SignOutActionList";
+import SettingsNav from "./SettingsNav";
+import FormMessage from "./FormMessage";
+import boxArrow from "./../box-arrow-right.svg";
+import errorIcon from "../../../component-assets/icons/bx-error.svg";
+import axios from "axios";
+
+function AllSessionSignOut() {
+  const [error, setError] = useState("");
+  //const [userSession, setUserSession] = useState({})
```
Comment (billmal071) [inline packages/main/src/components/sessions-signout/AllSessionSignOut.jsx]: Remove.

File: packages/main/src/components/sessions-signout/AllSessionSignOut.jsx [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+  const history = useHistory();
+
+  useEffect(() => {
+    if (
+      sessionStorage.getItem("user") &&
+      sessionStorage.getItem("session_id")
+    ) {
+      const userInfo = JSON.parse(sessionStorage.getItem("user"));
+      // console.log(userInfo)
```
Comment (billmal071) [inline packages/main/src/components/sessions-signout/AllSessionSignOut.jsx]: remove.

File: packages/main/src/components/sessions-signout/AllSessionSignOut.jsx [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+  const handleSubmit = async e => {
+    e.preventDefault();
+
+    const passwordData = {
+      password: currentPassword
+    };
+
+    try {
+      const response = await axios.post(
+        "https://api.zuri.chat/auth/logout/othersessions",
```
Comment (billmal071) [inline packages/main/src/components/sessions-signout/AllSessionSignOut.jsx:50]: There are different environments, use the environment variable for this.

File: packages/main/src/components/sessions-signout/AllSessionSignOut.jsx [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+        {
+          body: JSON.stringify(passwordData)
+        }
+      );
+      if (response.status !== 200) {
+        throw new Error(`Failed to fetch data, status: ${response.status} `);
```
Comment (billmal071) [inline packages/main/src/components/sessions-signout/AllSessionSignOut.jsx:56]: Use a toast to notify users of the error.

File: packages/main/src/components/sessions-signout/AllSessionSignOut.jsx [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+      }
+      const { data } = await response.data;
+      setUser(data);
+      // console.log(data)
+    } catch (error) {
+      console.error(error);
```
Comment (billmal071) [inline packages/main/src/components/sessions-signout/AllSessionSignOut.jsx:61]: remove.

File: packages/main/src/components/settings-nav/SettingsNav.jsx [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+import styles from "../styles/settingsNav.module.css";
+import { BiMenuAltRight } from "react-icons/bi";
+import { Link } from "react-router-dom";
+import zuri from "../assets/zuri.svg";
+import navImage from "../assets/navImage.png";
+
+const SettingsNav = () => {
+  const [menu, setMenu] = useState(false);
+  // const state = useContext(TopbarContext)
+  // const [active] = state.presence
```
Comment (billmal071) [inline packages/main/src/components/settings-nav/SettingsNav.jsx]: remove.

File: packages/main/src/components/form-message/ConfirmPassword.module.css [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+  -moz-transition: all 0.3s ease-out;
+  border: none;
+  border-radius: 3px;
+  outline: none;
+}
+
+/* .form_group label span {
+  display: inline-block;
+  font-weight: 600;
+} */
```
Comment (billmal071) [inline packages/main/src/components/form-message/ConfirmPassword.module.css]: remove.

File: packages/main/src/components/settings-nav/SettingsNav.jsx [hunk 8]
```diff
@@ -1,11 +1,11 @@
 import React, { useState, useContext } from "react";
-import { TopbarContext } from "../../../context/Topbar";
+// import { TopbarContext } from "../../../context/Topbar";
```
Comment (billmal071) [inline packages/main/src/components/settings-nav/SettingsNav.jsx]: remove.

File: packages/main/src/components/settings-nav/SettingsNav.module.css [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+  width: 50%;
+  max-width: 100%;
+  height: 100%;
+}
+
+.brand h3 {
+  font-family: "Lato";
+  font-size: 25px;
+  font-weight: 700;
+  /* padding-top: 5px; */
```
Comment (billmal071) [inline packages/main/src/components/settings-nav/SettingsNav.module.css]: remove.

File: packages/main/src/components/settings-tab/SettingsTab.jsx [hunk 10]
```diff
@@ -77,9 +77,11 @@ const SettingsTab = () => {
           </span>
         </div>
         <div className={styles.settingsright}>
-          <button onClick={handleSignOut} className={styles.signout}>
-            Sign out of all othe sessions
-          </button>
+          <Link to="/admin/settings/accountsProfile/sessions-signout">
+            <button onClick={handleSignOut} className={styles.signout}>
```
Comment (billmal071) [inline packages/main/src/components/settings-tab/SettingsTab.jsx:81]: Why is it a link and a button?

**Review by billmal071 — CHANGES_REQUESTED** [review]

File: packages/main/src/components/sessions-signout/AllSessionSignOut.jsx (hunk 1 above)
Comment (Chisquare7) [reply packages/main/src/components/sessions-signout/AllSessionSignOut.jsx]: Done

File: packages/main/src/components/sessions-signout/AllSessionSignOut.jsx (hunk 2 above)
Comment (Chisquare7) [reply packages/main/src/components/sessions-signout/AllSessionSignOut.jsx]: Done

File: packages/main/src/components/sessions-signout/AllSessionSignOut.jsx [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+        {
+          body: JSON.stringify(passwordData)
+        }
+      );
+      if (response.status !== 200) {
+        throw new Error(`Failed to fetch data, status: ${response.status} `);
+      }
+      const { data } = await response.data;
+      setUser(data);
+      // console.log(data)
```
[the remark this answers is not in the document]
Comment (Chisquare7) [reply packages/main/src/components/sessions-signout/AllSessionSignOut.jsx]: Done

File: packages/main/src/components/settings-nav/SettingsNav.jsx [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+      </div>
+
+      <div className={styles.rightnav}>
+        <div className={styles.pImage}>
+          <img src={navImage} alt="Profile" className={styles.profileImg} />
+          {/* {active ? (
+            <div className={styles.circles}></div>
+          ) : (
+            <div className={styles.circleAway}></div>
+          )} */}
```
[the remark this answers is not in the document]
Comment (Chisquare7) [reply packages/main/src/components/settings-nav/SettingsNav.jsx]: Done

File: packages/main/src/components/form-message/ConfirmPassword.module.css (hunk 7 above)
Comment (Chisquare7) [reply packages/main/src/components/form-message/ConfirmPassword.module.css]: Done

File: packages/main/src/components/settings-nav/SettingsNav.jsx (hunk 8 above)
Comment (Chisquare7) [reply packages/main/src/components/settings-nav/SettingsNav.jsx]: Done

File: packages/main/src/components/settings-tab/SettingsTab.jsx (hunk 10 above)
Comment (Chisquare7) [reply packages/main/src/components/settings-tab/SettingsTab.jsx]: Done

File: packages/main/src/components/sessions-signout/AllSessionSignOut.jsx (hunk 5 above)
Comment (Chisquare7) [reply packages/main/src/components/sessions-signout/AllSessionSignOut.jsx:61]: Done

File: packages/main/src/components/settings-nav/SettingsNav.jsx (hunk 6 above)
Comment (Chisquare7) [reply packages/main/src/components/settings-nav/SettingsNav.jsx]: Done

File: packages/main/src/components/sessions-signout/AllSessionSignOut.jsx (hunk 3 above)
Comment (Chisquare7) [reply packages/main/src/components/sessions-signout/AllSessionSignOut.jsx:50]: Please I don't understand by environment variable....can you please add more context?
@billmal071

Comment (billmal071) [reply packages/main/src/components/sessions-signout/AllSessionSignOut.jsx:50]: There's an environment variable to get the base url for the api depending on if it's on dev, staging or main(prod).

**Pull request merged**